backends/p4tools/modules/smith/common/scope.cpp

```diff
@@ -38,6 +38,13 @@ void P4Scope::addToScope(const IR::Node *node) {
         addLval(dv->type, dv->name.name);
     } else if (const auto *dc = node->to<IR::Declaration_Constant>()) {
         addLval(dc->type, dc->name.name, true);
+    } else if (const auto *td = node->to<IR::Type_Typedef>()) {
+        lScope->push_back(td);
+        if (td->name == "SecurityAssocIdUint_t") {
```

Comment:
One way to implement this is to check whether the referenced type is also a Type_Name, then resolve it with `getTypeByName`. 

Alternatively, you can try to extend getTypeByName to recurse when the retrieved type is another indirect type.